src/change_stream/mod.rs

```diff
@@ -107,6 +111,25 @@ pub struct ChangeStream {
 }
 
 impl ChangeStream {
+    pub(crate) fn new(
+        change_streams_spec: ChangeStreamSpecification,
+        session: Option<ClientSession>,
+        client: Client,
+    ) -> Self {
+        let cursor = Cursor::new(client.clone(), change_streams_spec.cursor_spec, session);
+        Self {
+            cursor,
+            pipeline: change_streams_spec.pipeline,
+            client,
+            target: change_streams_spec.target,
+            resume_token: None,
+            options: change_streams_spec.options.clone(),
+            selection_criteria: change_streams_spec.options.unwrap().selection_criteria,
```

Comment:
We definitely do not want to unwrap here; if the user doesn't specify any options, this will panic and crash the entire application. Instead, use `and_then` to return the selection criteria if there is one or `None` if there either the options or criteria are `None`:

```rust
selection_critiera: change_streams_spec.options.and_then(|opts| opts.selection_critiera),`
```

Reply:
fixed